src/client/WebhookClient.js

```diff
@@ -10,17 +11,48 @@ const Webhook = require('../structures/Webhook');
  */
 class WebhookClient extends BaseClient {
   /**
-   * @param {Snowflake} id The webhook's id
-   * @param {string} token Token of the webhook
+   * The data for the webhook client containing either an id and token or just a URL
+   * @typedef {Object} WebhookClientData
+   * @property {(WebhookClientDataIdWithToken|WebhookClientDataURL)} data The data for the webhook client
+   */
+
+  /**
+   * The data for the webhook client containing both an id and token
+   * @typedef {Object} WebhookClientDataIdWithToken
+   * @property {Snowflake} id The webhook's id
+   * @property {string} token Token of the webhook
+   */
+
+  /**
+   * The data for the webhook client containing a URL
+   * @typedef {Object} WebhookClientDataURL
+   * @property {string} url The full URL of the webhook
+   */
+
+  /**
+   * @param {WebhookClientData} data The data of the webhook
    * @param {ClientOptions} [options] Options for the client
    * @example
    * // Create a new webhook and send a message
-   * const hook = new Discord.WebhookClient('1234', 'abcdef');
+   * const hook = new Discord.WebhookClient({ id: '1234', token: 'abcdef' });
    * hook.send('This will send a message').catch(console.error);
    */
-  constructor(id, token, options) {
+  constructor(data, options) {
     super(options);
     Object.defineProperty(this, 'client', { value: this });
+    let { id, token } = data;
+
+    if ('url' in data) {
+      const url = data.url.match(
+        // eslint-disable-next-line no-useless-escape
+        /^https?:\/\/(?:canary|ptb)?\.?discord\.com\/api\/webhooks(?:\/v[0-9]\d*)?\/([^\/]+)\/([^\/]+)/i,
+      );
+
+      if (!url || url.length <= 1) throw new Error('WEBHOOK_URL_INVALID');
```

Comment:
```suggestion
      if (url?.length !== 2) throw new Error('WEBHOOK_URL_INVALID');
```
```suggestion
      if (url?.length < 2) throw new Error('WEBHOOK_URL_INVALID');
```
either or
can also stay the same prob doesn't matter much